plutus-pab/src/Cardano/Protocol/Socket/Mock/Client.hs

```diff
@@ -0,0 +1,191 @@
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE NamedFieldPuns    #-}
+{-# LANGUAGE TypeFamilies      #-}
+
+module Cardano.Protocol.Socket.Mock.Client where
```

Comment:
Needs some comments, at least a pointer to the non-mock client, and explain what the difference is